tiledb/sm/array_schema/array_schema.cc

```diff
@@ -570,14 +570,30 @@ Status ArraySchema::deserialize(ConstBuffer* buff) {
   RETURN_NOT_OK(buff->read(&capacity_, sizeof(uint64_t)));
 
   // Load coords filters
-  RETURN_NOT_OK(coords_filters_.deserialize(buff));
+  auto&& [st_coords_filters, coords_filters]{FilterPipeline::deserialize(buff)};
+  if (!st_coords_filters.ok()) {
+    return Status_ArraySchemaError("Cannot deserialize coords filters");
+  }
+  coords_filters_ = FilterPipeline(*coords_filters.value());
```

Comment:
You shouldn't need to explicitly call a constructor here. The assignment operator should work fine.